modules/basegl/glsl/img_channel_combine.frag

```diff
@@ -0,0 +1,53 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2013-2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * 
+ *********************************************************************************/
+
+#include "utils/structs.glsl"
+
+uniform ImageParameters outportParameters;
+
+uniform sampler2D inport0Color;
+uniform sampler2D inport1Color;
+uniform sampler2D inport2Color;
+uniform sampler2D inport3Color;
+
+uniform bool use_alpha_texture_ = true;
+uniform float alpha_ = 1.0;
+
+void main() {
+    vec2 texCoords = gl_FragCoord.xy * outportParameters.reciprocalDimensions;
+    float r = texture(inport0Color, texCoords).r;
+    float g = texture(inport1Color, texCoords).r;
+    float b = texture(inport2Color, texCoords).r;
+    float a = alpha_;
```

Comment:
use else-if for alpha

Reply:
or even better `float a = mix(alpha_, texture(inport3Color, texCoords).r, use_alpha_texture_);`